fetch.bs

```diff
@@ -5075,6 +5094,32 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
 </ol>
 
 
+<h3 id=tao-check>TAO check</h3>
+
+<p>To perform a <dfn id=concept-tao-check>TAO check</dfn> for a <var>request</var> and
+<var>response</var>, run these steps:
+
+<ol>
+ <li><p>If <var>request</var>'s <a for=request>timing allow failed flag</a> is set, then return
+ failure.
+
+ <li><p>If <var>request</var>'s <a for=request>response tainting</a> is "<code>basic</code>", then return success.
+
+ <li><p>Let <var>values</var> be the result of
+ <a for="header list">getting, decoding, and splitting</a>
+ `<code>Timing-Allow-Origin</code></a>` from <var>response</var>'s <a for=response>header list</a>.
+
+ <li><p>If <var>values</var> contains `<code>*</code>`, then return success.
+
+ <li><p>Let <var>serializedOrigin</var> be the result of <a>serializing a request origin</a> with
+ <var>request</var>.
```

Comment:
Is there a test for this scenario and browser bugs given that it sounded liked this was not yet how browsers behaved?

How are we going to integrate this with the resource timing specification?

Reply:
> Is there a test for this scenario and browser bugs given that it sounded liked this was not yet how browsers behaved?

I'll modify https://github.com/web-platform-tests/wpt/blob/master/resource-timing/crossorigin-sandwich-TAO.sub.html because it should no longer pass the TAO check with this change. And will also add a crossorigin sandwich which uses '*' in the header to ensure that it still passes. I imagine browser bugs should be filed once the spec change has landed?
 
> How are we going to integrate this with the resource timing specification?

We'll replace the timing allow check algorithm with a query into the response's flag.

Reply:
They can be filed now (having a brief heads up window is nice), and the test change can be created as well. Ideally we also have a PR ready for Resource Timing. That way we can land it all together and know that it integrates well.

Reply:
https://github.com/web-platform-tests/wpt/blob/master/resource-timing/crossorigin-sandwich-TAO.sub.html
https://bugs.chromium.org/p/chromium/issues/detail?id=1022816
https://bugzilla.mozilla.org/show_bug.cgi?id=1597808
https://bugs.webkit.org/show_bug.cgi?id=204375

Reply:
Also https://github.com/w3c/resource-timing/pull/218 (yes, still hand-wavy because what is a resource, but need to fix the use of 'resource' throughout in order to solve that problem).